src/util/typeAssertions.ts

```diff
@@ -41,29 +40,36 @@ export const isCollection = (
 export const isActiveSelection = (
   fabricObject?: FabricObject
 ): fabricObject is ActiveSelection => {
-  return !!fabricObject && fabricObject.isType('ActiveSelection');
+  return (
+    !!fabricObject &&
+    (fabricObject.constructor as typeof FabricObject).type === 'ActiveSelection'
+  );
 };
 
 export const isTextObject = (
   fabricObject?: FabricObject
 ): fabricObject is Text => {
   // we could use instanceof but that would mean pulling in Text code for a simple check
   // @todo discuss what to do and how to do
-  return !!fabricObject && fabricObject.isType('Text', 'IText', 'Textbox');
-};
-
-export const isPath = (fabricObject?: FabricObject): fabricObject is Path => {
-  // we could use instanceof but that would mean pulling in Text code for a simple check
-  // @todo discuss what to do and how to do
-  return !!fabricObject && fabricObject.isType('Path');
+  return (
+    !!fabricObject &&
+    ['Text', 'IText', 'Textbox'].some(
+      (type) => type === (fabricObject.constructor as typeof FabricObject).type
+    )
+  );
 };
 
 export const isInteractiveTextObject = (
   fabricObject?: FabricObject
 ): fabricObject is IText | Textbox => {
   // we could use instanceof but that would mean pulling in Text code for a simple check
   // @todo discuss what to do and how to do
-  return !!fabricObject && fabricObject.isType('IText', 'Textbox');
+  return (
+    !!fabricObject &&
```

Comment:
If I create a class TextBox2 that inherits from TextBox, it won't match here.  isPath and isTextObject also have the same issue. Would it be better to write it like this?
```
fabricObject instanceof IText && !(fabricObject instanceof Text)
```

Reply:
instanceof is the best solution not doubt
but it causes import cycles so we can't use it
this is part of the reason why I want to remove all these utils as done in #8950